src/meta.rs

```diff
@@ -12,6 +14,40 @@ pub struct SpotMeta {
     pub tokens: Vec<TokenInfo>,
 }
 
+impl SpotMeta {
+    pub fn add_pair_and_name_to_index_map(
+        &self,
+        mut coin_to_asset: HashMap<String, u32>,
+    ) -> HashMap<String, u32> {
+        let index_to_name: HashMap<usize, &str> = self
+            .tokens
+            .iter()
+            .map(|info| (info.index, info.name.as_str()))
+            .collect();
+
+        for asset in self.universe.iter() {
+            let spot_ind: u32 = 10000 + asset.index as u32;
+            let name_to_ind = (asset.name.clone(), spot_ind);
+
+            let token_1_name = if let Some(name) = index_to_name.get(&asset.tokens[0]) {
+                name
+            } else {
+                "" // Keeping these empty because the user won't be able to match then so no harm
```

Comment:
nit: just continue

also use `let/else` for this:
```
let Some(token1_name) = index_to_name.get(&asset.tokens[0]) else {
  continue;
};
```

Reply:
Good idea. Updated, thanks.